### PR TITLE
fix(table): fix column default rendering errors

### PR DESCRIPTION
**Pull request by justwiner**
When rendering columns by default, default parameters should not be provided. Default parameters cannot be passed in as props of other components, which will cause other components to report errors

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/table/src/table-column/index.ts [hunk 1]
```diff
@@ -209,20 +210,15 @@ export default defineComponent({
...
-          _self: {},
-          column: {},
-          row: {},
-          $index: -1,
-        }),
-      )
+      children = this.$slots
+        .default?.()
+        .filter(childNode => childNode.type?.name === 'ElTableColumn')
     } catch {
```
Comment (JeremyWuuuuu) [inline packages/table/src/table-column/index.ts:219]: You need to handle `v-for` when the root tag is `Fragment`, you have to filter at least 2 twice.

Comment (JeremyWuuuuu) [reply packages/table/src/table-column/index.ts:219]: For example, when use give you template like:
```vue
<el-table>
  <el-column v-for="" />
</el-table>
```

Comment (JeremyWuuuuu) [reply packages/table/src/table-column/index.ts:219]: The expression `this.$slots.default?.()` will resolve a value: `VNode<Fragment>` which only have length of 1 and its type will be `Fragment === childNode.type.name`

Comment (justwiner) [reply packages/table/src/table-column/index.ts]: yes, now it's fixed

**Review by Ryan2128 — APPROVED** [review]

**Pull request merged**